src/rules/no-extra-semicolons/index.js

```diff
@@ -51,14 +53,39 @@ export default function (actual) {
     root.walk((node) => {
       if (node.raws.before) {
         const rawBeforeNode = node.raws.before
+        let allowedSemi = 0
+
+        if (node.type === "rule" && isCustomPropertySet(node.selector)) {
+          allowedSemi = 1
+        }
+
+        let countSemi = 0
+
         styleSearch({ source: rawBeforeNode, target: ";" }, match => {
+          countSemi++
```

Comment:
Shouldn't `styleSearch` already return the info you're looking for here? https://github.com/davidtheclark/style-search#reporting-matches